test/suite/run.py

```diff
@@ -178,16 +178,38 @@ def show_env(verbose, envvar):
 # e.g. test_util03 -> util
 reCatname = re.compile(r"test_([^0-9]+)[0-9]*")
 
+# look for a list of the form 0-9,11,15-17
+def parse_int_list(str):
+    ret = {}
+    # divide into ranges separated by commas
+    for r in str.split(","):
+        # split the range
+        bounds = r.split("-")
+        if len(bounds) == 1 and bounds[0].isdigit():
+            # single number with no dash
+            scenario = int(bounds[0])
+            ret[scenario] = True
```

Comment:
The choice to use a dictionary here is to speed up the lookup later on, correct?

Reply:
Yes. I mean, it doesn't really matter, but might as well.

Reply:
Can you add a comment on line 183 stating the reason behind that choice? It's fairly obvious but I'd like it be explicit.

Reply:
Sure.

Reply:
Thanks, that's what I expected.